docs/documentation/tracking-uploads.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Tracking Uploads
+id: tracking-uploads
+slug: /tracking-uploads
+sidebar_label: Tracking Uploads
+---
+
+You can track the progress of your uploads using "tags". Each tag tracks how many chunks were **split**, **stored**, **seen**, and **synced** by the network. By creating a tag before uploading and passing it to the upload function, you make the upload *trackable, allowing you to confirm whether your uploaded data has been fully synced.
+
+## How It Works
+
+### 1. Create a Tag
+
+Before uploading, create a new tag using `bee.createTag()`. This returns a unique tag UID that will be used to monitor the upload.
+
+```js
+const tag = await bee.createTag()
+console.log("Created new tag with UID:", tag.uid)
+```
+
+Alternatively, you can use an existing tag from `bee.getAllTags()` (useful for testing or reuse):
+
+```js
+const allTags = await bee.getAllTags()
+if (allTags.length > 0) {
+  tag = allTags[0]
+  console.log("Using existing tag with UID:", tag.uid)
+}
+```
+
+### 2. Upload a File with the Tag
+
+To enable tracking, pass the tag UID into the upload options under the `tag` key:
+
+```js
+const result = await bee.uploadFile(postageBatchId, fileData, 'nodes.json', {
+  tag: tag.uid,
+  contentType: 'application/json'
+})
+```
+
+This links the upload to your tag so you can monitor its progress.
+
+### 3. Track Tag Progress
+
+Use `bee.retrieveTag(tagUid)` to check how many chunks have been split and how many are synced. Poll repeatedly until `synced === split` to know when the upload has fully propagated.
```

Comment:
Correctly: `synced + seen === split`

"Seen" are chunks which were synced previously and therefor won't be this time.

Reply:
Ahh, so I suppose that means `seen` is only relevant for a re-upload?

Reply:
Or if there are common chunks which have appeared in previous files (such as zero byte paddings)